feat(i18n): pages/markdown-content to pt-br (#551)
withastro/docs: src/pages/pt-BR/guides/markdown-content.md

```markdown
---
layout: ../layouts/BaseLayout.astro
---
```

Um layout típico para páginas Markdown inclui:

1. a prop `content` para acessar os dados do frontmatter da página Markdown.
2. um [`<slot />`](/pt-BR/core-concepts/astro-components#slots) default para indicar onde o conteúdo Markdown da página deve ser renderizado.

```astro
---
// src/layouts/BaseLayout.astro
// 1. A prop content dá acesso aos dados do frontmatter
const { content } = Astro.props;
---
<html>
  <head>
    <!-- Adicione outros elementos Head aqui, como estilos e meta tags. -->
    <title>{content.title}</title>
  </head>
  <body>
    <!-- Adicione outros componentes de UI aqui, como cabeçalhos e rodapés comuns. -->
    <h1>{content.title} por {content.author}</h1>
    <!-- 2. O HTML renderizado será passado para o slot default. -->
    <slot />
    <p>Escrito em: {content.date}</p>
  </body>
</html>
```

A prop `content` também contém uma propriedade `astro` com metadados adicionais sobre a página, como o objeto Markdown `source` completo e um objeto `headers`.

Um exemplo de objeto `content` de post de blog pode ser algo como:

```json
{

>   /** Frontmatter from a blog post
>   "title": "Lançamento do Astro 0.18",
>   "date": "Terça-feira, 27 de julho de 2021",
>   "author": "Matthew Phillips",
>   "description": "Astro 0.18 é o nosso maior lançamento desde o lançamento do Astro.",
>   "draft": false,
>   "keywords": ["astro", "lancamento", "anuncio"]

Reviewer: Adding the missing translations here. `draft` isn't translated because it makes part of Astro itself.
```suggestion
  /** Frontmatter de uma postagem de blog
  "titulo": "Lançamento do Astro 0.18",
  "data": "Terça-feira, 27 de julho de 2021",
  "autor": "Matthew Phillips",
  "descricao": "Astro 0.18 é o nosso maior lançamento desde o lançamento do Astro.",
  "draft": false,
  "palavras-chave": ["astro", "lancamento", "anuncio"]
```

</Biography>
```

## Importar Markdown

Você pode importar arquivos Markdown diretamente em seus arquivos Astro! Você pode importar uma página específica com `import` ou várias com `Astro.glob()`.

```astro
---
// Importe markdown. import() dinâmico também é suportado!
import * as greatPost from '../pages/post/great-post.md';

// Você também pode importar múltiplos arquivos com Astro.glob
const posts = await Astro.glob('../pages/post/*.md');
---

Ótimo post: <a href={greatPost.url}>{greatPost.frontmatter.title}</a>

<ul>
  {posts.map(post => <li>{post.frontmatter.title}</li>)}
</ul>
```

Cada arquivo Markdown exporta as seguintes propriedades:

- `frontmatter`: Quaisquer dados especificados no frontmatter YAML deste arquivo.
- `file`: O caminho absoluto deste arquivo (e.g. `/home/user/projects/.../file.md`).
- `url`: Se é uma página, o URL da página (e.g. `/pt-BR/guides/markdown-content`).
- `getHeaders()`: Uma função assíncrona que retorna os cabeçalhos do arquivo Markdown. A resposta é desse tipo: `{ depth: number; slug: string; text: string }[]`.
- `Content`: Um componente que renderiza o conteúdo do arquivo Markdown. Eis um exemplo:

  ```astro
  ---

>   import {Content as PromoBanner} from '../components/promoBanner.md';
>   ---
> 
>   <h2>Promoção de hoje</h2>
>   <PromoBanner />

Reviewer: ```suggestion
  import {Content as BannerPromocional} from '../components/bannerPromocional.md';
  ---
  
  <h2>Promoção de hoje</h2>
  <BannerPromocional />
```

  ```

Opcionalmente, você pode fornecer um tipo para a variável `frontmatter` usando um genérico TypeScript:

```astro
---
interface Frontmatter {
  title: string;
  description?: string;
}

> const posts = await Astro.glob<Frontmatter>('../pages/post/*.md');
> ---
> 
> <ul>
>   {posts.map(post => <li>{post.title}</li>)}
>   <!-- post.title vai ser `string`! -->
> </ul>

Reviewer: ```suggestion
const postagens = await Astro.glob<Frontmatter>('../pages/postagens/*.md');
---

<ul>
  {postagens.map(postagem => <li>{postagem.titulo}</li>)}
  <!-- postagem.titulo vai ser `string`! -->
</ul>
```

```

## Componente Markdown

> > NOTA: O componente `<Markdown />` não funciona em SSR e pode ser removido antes da v1.0. Se possível, deve ser evitado. Para usar Markdown em seus templates, use um arquivo `.md` separado e então [`import` Markdown](/pt-BR/guides/markdown-content#importing-markdown) no seu template como um componente.

Reviewer: ```suggestion
> NOTA: O componente `<Markdown />` não funciona em SSR e pode ser removido antes da v1.0. Se possível, deve ser evitado. Para usar Markdown em seus templates, use um arquivo `.md` separado e então [`import` Markdown](/pt-BR/guides/markdown-content#importar-markdown) no seu template como um componente.
```

Você pode importar o [componente Markdown nativo do Astro](/pt-BR/reference/api-reference#markdown-) em seu script de componente e, em seguida, escrever qualquer Markdown que quiser entre as tags `<Markdown></Markdown>`.

````astro
---
import { Markdown } from 'astro/components';
import Layout from '../layouts/Layout.astro';

> const expressions = 'Lorem ipsum';
> ---
> <Layout>
>   <Markdown>
>     # Olá, mundo!
> 
>     **Tudo** suportado em um arquivo `.md` também é suportado aqui!
> 
>     Com _zero_ sobrecarga de tempo de execução.
> 
>     Além disso, o Astro suporta:
>     - {expressões} Astro
>     - Normalização automática de indentação
>     - Escaping automático de expressões dentro de blocos de código
> 
>     ```js
>       // Este conteúdo não é transformado!
>       const object = { someOtherValue };

Reviewer: ```suggestion
const expressoes = 'Lorem ipsum';
---
<Layout>
  <Markdown>
    # Olá, mundo!

    **Tudo** que é suportado em um arquivo `.md` também é suportado aqui!

    Com _zero_ sobrecarga em tempo de execução.

    Além disso, o Astro suporta:
    - {expressoes} Astro
    - Normalização automática de indentação
    - Escapamento automático de expressões dentro de blocos de código

    ```js
      // Este conteúdo não é transformado!
      const objeto = { algumOutroValor };
```

    ```

    - Suporte a componentes ricos, como em qualquer arquivo `.astro`!
    - Suporte a Markdown recursivo (os filhos do Componente também são processados como Markdown)
  </Markdown>
</Layout>
````

### Markdown Remoto

> NOTA: O componente `<Markdown />` não funciona em SSR e pode ser removido antes da v1.0. Se possível, deve ser evitado. Para usar Markdown em seus templates, use um arquivo `.md` separado e dê `import` no seu template como um componente. Leia esta [Discussão RFC](https://github.com/withastro/rfcs/discussions/179) para saber mais.

Se você tem Markdown em uma fonte remota, você pode passá-lo diretamente para o componente Markdown através do atributo `content`.

```astro
---
import { Markdown } from 'astro/components';

> const content = await fetch('https://raw.githubusercontent.com/withastro/docs/main/README.md').then(res => res.text());
> ---
> <Layout>
>   <Markdown content={content} />

Reviewer: ```suggestion
const conteudo = await fetch('https://raw.githubusercontent.com/withastro/docs/main/README.md').then(res => res.text());
---
<Layout>
  <Markdown content={conteudo} />
```

</Layout>
```

### Markdown Aninhado

> NOTA: O componente `<Markdown />` não funciona em SSR e pode ser removido antes da v1.0. Se possível, deve ser evitado. Para usar Markdown em seus templates, use um arquivo `.md` separado e dê `import` no seu template como um componente. Leia esta [Discussão RFC](https://github.com/withastro/rfcs/discussions/179) para saber mais.

Componentes `<Markdown />` podem ser aninhados.

```astro
---
import { Markdown } from 'astro/components';

> const content = await fetch('https://raw.githubusercontent.com/withastro/docs/main/README.md').then(res => res.text());
> ---
> 
> <Layout>
>   <Markdown>
>     ## Exemplo de Markdown
> 
>     Aqui temos algum código __Markdown__. Também podemos renderizar conteúdo remoto dinamicamente.
> 
>     <Markdown content={content} />

Reviewer: ```suggestion
const conteudo = await fetch('https://raw.githubusercontent.com/withastro/docs/main/README.md').then(res => res.text());
---

<Layout>
  <Markdown>
    ## Exemplo de Markdown

    Aqui temos algum código __Markdown__. Também podemos renderizar conteúdo remoto dinamicamente.

    <Markdown content={conteudo} />
```

  </Markdown>
</Layout>
```

> ⚠️ O uso do componente `Markdown` para renderizar Markdown remoto pode abrir brecha para um ataque [cross-site scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting). Se você estiver renderizando conteúdo não confiável, certifique-se de _sanitizar seu conteúdo **antes** de renderizá-lo_.

Reviewer: Added the pt-BR Wikipedia link.
```suggestion
⚠️ O uso do componente `Markdown` para renderizar Markdown remoto pode abrir brecha para um ataque de [cross-site scripting (XSS)](https://pt.wikipedia.org/wiki/Cross-site_scripting). Se você estiver renderizando conteúdo não confiável, certifique-se de _sanitizar seu conteúdo **antes** de renderizá-lo_.
```

   // astro.config.mjs
   export default {
     markdown: {
       remarkPlugins: [
          // Adicione um plugin Remark que você deseja habilitar para seu projeto.
          // Se você precisar fornecer opções para o plugin, você pode usar um array e colocar as opções como o segundo item.
          // ['remark-autolink-headings', { behavior: 'prepend'}],
       ],
       rehypePlugins: [
          // Adicione um plugin Rehype que você deseja habilitar para seu projeto.
          // Se você precisar fornecer opções para o plugin, você pode usar um array e colocar as opções como o segundo item.
          // 'rehype-slug',
          // ['rehype-autolink-headings', { behavior: 'prepend'}],
       ],
     },
   };
   ```

    Você pode informar os nomes dos plugins, bem como importá-los:    

   ```js
   // astro.config.mjs
   import autolinkHeadings from 'remark-autolink-headings';

   export default {
     markdown: {
       remarkPlugins: [[autolinkHeadings, { behavior: 'prepend' }]],
     },
   };
   ```

### Syntax Highlighting

Astro vem com suporte nativo para [Shiki](https://shiki.matsu.io/) e [Prism](https://prismjs.com/). Isso fornece syntax highlighting imediato para:

> - todas as code fences (\`\`\`) usadas em um arquivo markdown (`.md`) e o [componente `<Markdown />` nativo](#markdown-component).

Reviewer: ```suggestion
- todas as code fences (\`\`\`) usadas em um arquivo markdown (`.md`) e o [componente `<Markdown />` nativo](#componente-markdown).
```

> ### Syntax Highlighting
> 
> Astro vem com suporte nativo para [Shiki](https://shiki.matsu.io/) e [Prism](https://prismjs.com/). Isso fornece syntax highlighting imediato para:
> 
> - todas as code fences (\`\`\`) usadas em um arquivo markdown (`.md`) e o [componente `<Markdown />` nativo](#markdown-component).
> - conteúdo dentro do [componente `<Code />` nativo](/pt-BR/reference/api-reference/#code-) (oferecido por Shiki) ou o [componente `<Prism />`](/pt-BR/reference/api-reference/#prism-) (oferecido por Prism).
> 
> Shiki é ativado por padrão, pré-configurado com o tema `github-Dark`. A saída compilada será limitada a `style`s inline sem classes CSS de fora, folhas de estilo ou JS client-side.
> 
> Se você optar por usar Prism, aplicaremos as classes CSS do Prism. Observe que **você precisa colocar sua própria folha de estilo CSS** para o syntax highlighting funcionar! Veja a [seção de configuração do Prism](#prism-configuration) para mais detalhes.

Reviewer: ```suggestion
### Destaque de Sintaxe

Astro vem com suporte nativo para [Shiki](https://shiki.matsu.io/) e [Prism](https://prismjs.com/). Isso fornece destaque de sintaxe imediato para:

- todas as cercas de código (\`\`\`) usadas em um arquivo Markdown (`.md`) e ao [componente `<Markdown />` nativo](#markdown-component).
- conteúdo dentro do [componente `<Code />` nativo](/pt-BR/reference/api-reference/#code-) (oferecido por Shiki) ou o [componente `<Prism />`](/pt-BR/reference/api-reference/#prism-) (oferecido por Prism).

Shiki é ativado por padrão, pré-configurado com o tema `github-dark`. A saída compilada será limitada a `style`s inline sem classes CSS de fora, folhas de estilo ou JS no lado do cliente.

Se você optar por usar Prism, aplicaremos as classes CSS do Prism. Observe que **você precisa colocar sua própria folha de estilo CSS** para o destaque de sintaxe funcionar! Veja a [seção de configuração do Prism](#configuração-prism) para mais detalhes.
```

> Se você optar por usar Prism, aplicaremos as classes CSS do Prism. Observe que **você precisa colocar sua própria folha de estilo CSS** para o syntax highlighting funcionar! Veja a [seção de configuração do Prism](#prism-configuration) para mais detalhes.

Reviewer: ```suggestion
Se você optar por usar Prism, aplicaremos as classes CSS do Prism. Observe que **você precisa colocar sua própria folha de estilo CSS** para o syntax highlighting funcionar! Veja a [seção de configuração do Prism](#configuração-prism) para mais detalhes.
```

> #### Escolha um syntax highlighter
> 
> Shiki é o nosso syntax highlighter default. Se você quiser mudar para `'prism'` ou desativar completamente o syntax highlighting, você pode usar o objeto de configuração `markdown`:

Reviewer: ```suggestion
#### Escolha um destacador de sintaxe

Shiki é o nosso destacador de sintaxe padrão. Se você quiser mudar para `'prism'` ou desativar completamente o destaque de sintaxe, você pode usar o objeto de configuração `markdown`:
```

```js
// astro.config.mjs
export default {
  markdown: {

>     // Pode ser 'shiki' (default), 'prism' ou false para desabilitar o highlighting

Reviewer: ```suggestion
   // Pode ser 'shiki' (padrão), 'prism' ou false para desabilitar o destacamento
```

    syntaxHighlight: 'prism',
  },
};
```

#### Configuração Shiki

Ao usar Shiki, você pode configurar todas as opções por meio do objeto `shikiConfig`, tal como:

```js
// astro.config.mjs
export default {
  markdown: {
    shikiConfig: {
      // Escolha os temas internos do Shiki (ou adicione o seu)
      // https://github.com/shikijs/shiki/blob/main/docs/themes.md
      theme: 'dracula',
      // Adicione idiomas personalizados

>       // Nota: Shiki tem inúmeras langs nativas, incluindo .astro!
>       // https://github.com/shikijs/shiki/blob/main/docs/languages.md
>       langs: [],
>       // Habilite quebra de linha para evitar scroll horizontal

Reviewer: ```suggestion
      // Nota: Shiki tem inúmeras linguagens nativas, incluindo .astro!
      // https://github.com/shikijs/shiki/blob/main/docs/languages.md
      langs: [],
      // Habilite quebra de linha para evitar rolagem horizontal
```

      wrap: true,
    },
  },
};
```

> Também sugerimos [inspecionar a documentação de tema deles](https://github.com/shikijs/shiki/blob/main/docs/themes.md#loading-theme) para explorar o carregamento de tema personalizado, light mode vs dark mode ou estilizar via variáveis CSS.

Reviewer: ```suggestion
Também sugerimos [inspecionar a documentação de tema deles](https://github.com/shikijs/shiki/blob/main/docs/themes.md#loading-theme) para explorar o carregamento de tema personalizado, modo claro vs modo escuro ou estilizar via variáveis CSS.
```

#### Configuração Prism

> Ao usar o Prism, você precisará adicionar uma folha de estilo ao seu projeto para syntax highlighting. Se você acabou de começar e prefere usar Prism em vez de Shiki, sugerimos:
> 
> 1. [Colocar `syntaxHighlight: 'prism'`](#choose-a-syntax-highlighter) na sua configuração `@astrojs/markdown-remark`.
> 2. Escolher uma folha de estilo predefinida entre os [Temas Prism](https://github.com/PrismJS/prism-themes) disponíveis.
> 3. Adicionar essa folha de estilo no [diretório `public/` do seu projeto](/pt-BR/core-concepts/project-structure/#public).
> 4. Carregá-la [no `<head>` de sua página](/pt-BR/core-concepts/astro-pages/#page-html) através de uma tag `<link>`.

Reviewer: ```suggestion
Ao usar o Prism, você precisará adicionar uma folha de estilo ao seu projeto para destacamento de sintaxe. Se você acabou de começar e prefere usar Prism em vez de Shiki, sugerimos:

1. [Colocar `syntaxHighlight: 'prism'`](#escolha-um-destacador-de-sintaxe) na sua configuração `@astrojs/markdown-remark`.
2. Escolher uma folha de estilo predefinida entre os [Temas Prism](https://github.com/PrismJS/prism-themes) disponíveis.
3. Adicionar essa folha de estilo no [diretório `public/` do seu projeto](/pt-BR/core-concepts/project-structure/#public).
4. Carregá-la [no `<head>` de sua página](/pt-BR/core-concepts/astro-pages/#html-da-página) através de uma tag `<link>`.
```

> 4. Carregá-la [no `<head>` de sua página](/pt-BR/core-concepts/astro-pages/#html-da-pagina) através de uma tag `<link>`.

Reviewer: ```suggestion
4. Carregá-la [no `<head>` de sua página](/pt-BR/core-concepts/astro-pages/#html-da-página) através de uma tag `<link>`.
```
